docs/notebooks/burger.ipynb

```diff
@@ -0,0 +1,745 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# *checkpoint_schedule* application: Adjoint-Based Gradient with the Burger's Equation\n",
+    "\n",
+    "This example shows adjoint-based gradient computation using the *checkpointing_schedules* package. We initially define the adjoint-based gradient problem and then present the forward and adjoint solvers prescribed by the *checkpointing_schedules* package.\n",
+    "\n",
+    "## Defining the application\n",
+    "\n",
+    "Let us consider a one-dimensional (1D) problem aiming to compute the gradient/sensitivity of the kinetic energy at a time $t=\\tau$ with respect to an initial condition $u_0$. Then, our functional can be expressed as:\n",
+    "$$\n",
+    "I(u(x, \\tau, u_0)) = \\int_\\Omega \\frac{1}{2} u(x, \\tau, u_0)u(x, \\tau, u_0) \\, dx.\n",
+    "\\tag{1} \n",
+    "$$\n",
+    "\n",
+    "In the current case, the velocity variable $u = u(x, t)$ is governed by the 1D viscous Burgers equation, a non-linear equation for the advection and diffusion of momentum:\n",
+    "\n",
+    "$$\n",
+    "\\frac{\\partial u}{\\partial t} + u \\frac{\\partial u}{\\partial x} - \\nu \\frac{\\partial^2 u}{\\partial x^2} = 0.\n",
+    "\\tag{2}\n",
+    "$$\n",
+    "\n",
+    "Here, $x \\in [0, L]$ is the space variable, and $t \\in \\mathbb{R}^{+}$ represents the time variable. The boundary condition is $u(0, t) = u(L, t) = 0$, where $L$ is the length of the 1D domain. The initial condition is given by $u_0 = \\sin(\\pi x)$.\n",
+    "\n",
+    "The control parameter is the initial condition $u_0$. Hence, the objective is to compute the adjoint-based gradient of the functional $I(u_0)$ with respect to $u_0$.\n",
+    "\n",
+    "This example uses the discrete adjoint formulation, meaning the adjoint system is derived after discretization of the forward problem."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adjoint-based gradient\n",
+    "We seek to compute the sensitivity/gradient $\\mathrm{d}\\widehat{I}/\\mathrm{d}u_0$, where $\\widehat{I} = I(u_0)$ is the reduced functional. On applying the chain rule, we have the below expression on considering the functional $I$ as in Eq. (1).\n",
+    "$$\\frac{\\mathrm{d}\\widehat{I}}{\\mathrm{d}u_0} = \\frac{\\partial I}{\\partial u} \\frac{\\mathrm{d}u}{\\mathrm{d}u_0}. \\tag{3}$$\n",
+    "Computing $\\partial J/\\partial u$ is straightforward since the functional (Eq. (1)) is written in terms of the solution variable $u$. In contrast, the Jacobian $\\mathrm{d}u/\\mathrm{d}u_0$ involves an expensive computational operation that is avoided by using the adjoint method.\n",
+    "\n",
+    "Taking the adjoint (Hermitian transpose) of the above equation:\n",
```

Comment:
Missing an equation? Suggest citing  https://doi.org/10.1137/1.9780898718720 section 2.3.3, which looks like the same derivation.

Reply:
By mistake, I removed texts on the latest commits. This is now fixed. The reference  https://doi.org/10.1137/1.9780898718720  was added. I verified the book section 2.3.3.